lib/main.dart

```diff
@@ -276,93 +276,86 @@ class _ThaliAppState extends State<ThaliApp> {
-                          BlocProvider(
-                            create: (_) => FullMemberCubit(
-                              authState.apiRepository,
-                            )..load(),
-                            lazy: false,
-                          ),
-                          BlocProvider(
-                            create: (_) => WelcomeCubit(
-                              authState.apiRepository,
-                            )..load(),
-                            lazy: false,
-                          ),
-                          BlocProvider(
-                            create: (_) => CalendarCubit(
-                              authState.apiRepository,
-                            )..cachedLoad(),
-                            lazy: false,
-                          ),
-                          BlocProvider(
-                            create: (_) => MemberListCubit(
-                              authState.apiRepository,
-                            )..load(),
-                            lazy: false,
-                          ),
-                          BlocProvider(
-                            create: (_) => AlbumListCubit(
-                              authState.apiRepository,
-                            )..load(),
-                            lazy: false,
-                          ),
-                          BlocProvider(
-                            // The SettingsCubit must not be lazy, since
-                            // it handles setting up push notifications.
-                            create: (_) => SettingsCubit(
-                              authState.apiRepository,
-                            )..load(),
-                            lazy: false,
-                          ),
-                          BlocProvider(
-                            create: (_) => TostiAuthCubit()..load(),
-                            lazy: true,
-                          ),
-                          BlocProvider(
-                            create: (_) => BoardsCubit(
-                              authState.apiRepository,
-                            )..load(),
-                            lazy: true,
-                          ),
-                          BlocProvider(
-                            create: (_) => CommitteesCubit(
-                              authState.apiRepository,
-                            )..load(),
-                            lazy: true,
-                          ),
-                          BlocProvider(
-                            create: (_) => SocietiesCubit(
-                              authState.apiRepository,
-                            )..load(),
-                            lazy: true,
-                          ),
-                        ],
-                        child: navigator!,
+                  if (authState is LoggedInAuthState ||
+                      (authState is LoggedOutAuthState &&
+                          authState.apiRepository != null)) {
```

Comment:
Actually this is a trick i did to get rid of the ugly 'no navigator around loginscreen' trick to prevent providernotfounderrors

Reply:
Still needs some attention perhaps, but the no navigator trick had to go as it doesnt allow showing dialogs there

Reply:
It would be nice to update the comment on how it works, as it is outdated atm. It would be nice to have documentation on how/why this works

Reply:
Or leave it to a different PR as I don't think it is necessary for this PR(?)

Reply:
But I'll look at making it a bit cleaner and documenting it

Reply:
I documented it a bit better. One option for later might be to instead modify the ConcrexitApiRepository to depend on a provided Client: we always provide ApiRepository, but sometimes that doesn't have a client and hence always raises 'LoggedOutException' or so. (and LoggedInAuthState contains a Client instead of ApiRepository).

But that also kind of just moves the problem elsewhere.

For now I think this is at least more elegant than the hack we had before.